pkgs/applications/video/natron/default.nix

```diff
@@ -0,0 +1,107 @@
+{ stdenv, fetchgit, fetchurl, qt4, pkgconfig, expat, cairo, boost, glew,
+  openexr, ffmpeg, opencolorio, openimageio, imagemagick }:
+
+let
+  version = "1.2.1";
+  CImg = fetchurl {
+    url = https://raw.githubusercontent.com/dtschump/CImg/d89fbcf399b3663ce3ebac2ee387642fd64014b6/CImg.h;
+    sha256 = "065sbx8dkywkrimc0l0vpq7dy7m9y6jlaq84x5ngys661j2nnslv";
+  };
+  plugins = {
+    misc = fetchgit {
```

Comment:
If there are submodules in a repository then use `fetchgit` otherwise use `fetchFromGithub` :)